src/Stackage/Database/Cron.hs

```diff
@@ -218,10 +218,11 @@ runStackageUpdate doNotUpload = do
     runStackageMigrations
     didUpdate <- forceUpdateHackageIndex (Just "stackage-server cron job")
     case didUpdate of
-        UpdateOccurred -> do
-            logInfo "Updated hackage index. Getting deprecated info now"
-            getHackageDeprecations >>= run . mapM_ addDeprecated
+        UpdateOccurred -> logInfo "Updated hackage index"
         NoUpdateOccurred -> logInfo "No new packages in hackage index"
+    logInfo "Getting deprecated info now"
+    deprecatedIds <- run . mapM addDeprecated =<< getHackageDeprecations
```

Comment:
This will hold onto presumably a large number of IDs in memory, and it will grow linearly over time. What about instead deleting all of the deprecated entries and then re-adding them? Since it all occurs in a single database transaction, this should be atomic from an end-user standpoint.

Reply:
That is a good idea. I've now changed it to removal then readding in a single sql transaction